File: src/main/java/by/logvin/mip/service/DrugService.java
package by.logvin.mip.service;

import by.logvin.mip.model.entity.Drug;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public interface DrugService {
    Drug findById(Long id);

    Page<Drug> findPaginated(Pageable pageable);

    Page<Drug> findByName(Long pharmacyId, String name, Pageable pageable);

    Page<Drug> findByStorage(Long storageId, Pageable pageable);

    Drug save(Drug drug);

    void delete(Long id);

    Drug update(Long id, Drug drug);

    Page<Drug> findAllDrugsByPharmacy(Long id, Pageable pageable);
}

File: front/src/services/drug/DrugService.js
import axios from "axios";


const DRUG_BASE_URL = "http://localhost:8080/drugs";

class DrugService {
    findById = async (drugId) => {
        return axios.get(`${DRUG_BASE_URL}/${drugId}`);
    }

    findAll = async (pharmacyId, name, page = 1, size = 10) => {
        return axios.get(`http://localhost:8080/pharmacies/${pharmacyId}/drugs?page=${page}&size=${size}` + (name ? `&name=${name}` : ""));
    }

    deleteById = async (drugId) => {
        return axios.delete(`${DRUG_BASE_URL}/${drugId}`);
    }

    save = async (drug) => {
        return axios.post(`${DRUG_BASE_URL}`, drug);
    }

    update = async (drugId, drug) => {
        return axios.put(`${DRUG_BASE_URL}/${drugId}`, drug);
    }
}

export default new DrugService();
